code-challenges.test.js

```diff
@@ -13,40 +13,134 @@
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
 // a) Create a test with expect statements using the variables provided.
-
+describe("gotString", () => {
+    it(" takes in a string and returns a string with a coded message ", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(gotString(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(gotString(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(gotString(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+// ReferenceError: gotString is not defined
+// b) Create the function that makes the test pass.
+// pseudocode - 
+// output -> takes in a string
+// input -> returns a string with a coded message which converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// process -
+// create a function named 'gotString'
+// using the .replace method we can replace the given letters/integers in the original string ->  str1.replace(regexp, str2)
+// the callback function (match) => codeMap[match] is used to change each matched letter to its corresponding code from the codeMap object
+// Method 1: 
```

Comment:
Based on your pseudo code, it seems like you don't fully understand what is happening with this solution.
- make an object with key value pairs representing letters to match and their replacement
- use .replace to iterate through string
- use regex with the letters to look for, using the g flag to find all occurrences, not just the first
- callback function will find each match and replace the matched letter with the value in the object